Assets/MixedRealityToolkit.SDK/Experimental/Features/UX/MixedRealityKeyboardBase.cs

```diff
@@ -0,0 +1,324 @@
+
+        private Coroutine stateUpdate;
+#endif
+
+        private bool multiLine = false;
+
+        #endregion Private fields
+
+        #region MonoBehaviour Implementation
+
+        /// <summary>
+        /// Initializes the UWP input pane.
+        /// </summary>
+        protected virtual void Start()
+        {
+ #if WINDOWS_UWP
+            UnityEngine.WSA.Application.InvokeOnUIThread(() =>
+            {
+                inputPane = InputPane.GetForCurrentView();
+                inputPane.Hiding += (inputPane, args) => OnKeyboardHiding();
+                inputPane.Showing += (inputPane, args) => OnKeyboardShowing();
+            }, false);
+#endif
+        }
+
+#if WINDOWS_UWP
+        private IEnumerator UpdateState()
+        {
+            while (true)
+            {
+                switch (state)
+                {
+                    case KeyboardState.Showing:
+                        {
+                            UpdateText();
+                        }
+                        break;
+
+                    case KeyboardState.Hiding:
+                        {
+                            if (onHideKeyboard != null)
+                            {
+                                onHideKeyboard.Invoke();
+                            }
+                        }
+                        break;
+                }
+
+                yield return null;
+            }
+        }
+#endif
+
+        private void OnDisable()
+        {
+            HideKeyboard();
+        }
+
+#endregion MonoBehaviour Implementation
+
+        /// <summary>
+        /// Closes the keyboard for user interaction.
+        /// </summary>
+        public void HideKeyboard()
+        {
+            if (state != KeyboardState.Hidden)
+            {
+                state = KeyboardState.Hidden;
+            }
+
+#if WINDOWS_UWP
+            UnityEngine.WSA.Application.InvokeOnUIThread(() => inputPane?.TryHide(), false);
+
+            if (stateUpdate != null)
+            {
+                StopCoroutine(stateUpdate);
+                stateUpdate = null;
+            }
+#endif
+        }
+
+        /// <summary>
+        /// Opens the keyboard for user interaction.
+        /// </summary>
+        /// <param name="text">Initial text to populate the keyboard with.</param>
+        /// <param name="multiLine">True, if the return key should signal a newline rather than a commit.</param>
+        public virtual void ShowKeyboard(string text = "", bool multiLine = false)
+        {
+            Text = text;
+            this.multiLine = multiLine;
+
+            // 2019/08/14: We show the keyboard even when the keyboard is already visible because on HoloLens 1
+            // and WMR the events OnKeyboardShowing and OnKeyboardHiding do not fire
+            //if (state == KeyboardState.Showing)
+            //{
+            //    Debug.Log($"MixedRealityKeyboard.ShowKeyboard called but keyboard already visible.");
+            //    return;
+            //}
```

Comment:
@julenka do you have any idea what this is? Since its fully commented out, remove?